avm/res/network/vpn-gateway/main.bicep

```diff
@@ -231,3 +231,74 @@ type routingConfigurationType = {
     }?
   }?
 }
+
+@export()
+@description('The type of VPN connection for VPN Gateway.')
+type vpnConnectionType = {
+  @description('Required. The name of the VPN connection.')
+  name: string
+  
+  @description('Optional. Connection bandwidth in MBPS.')
+  connectionBandwidth: int?
+  
+  @description('Optional. Enable BGP flag.')
+  enableBgp: bool?
+  
+  @description('Optional. Enable internet security.')
+  enableInternetSecurity: bool?
+  
+  @description('Optional. Remote VPN site resource ID.')
+  remoteVpnSiteResourceId: string?
+  
+  @description('Optional. Enable rate limiting.')
+  enableRateLimiting: bool?
+  
+  @description('Optional. Routing configuration.')
+  routingConfiguration: routingConfigurationType?
+  
+  @description('Optional. Routing weight.')
+  routingWeight: int?
+  
+  @description('Optional. Shared key.')
+  sharedKey: string?
+  
+  @description('Optional. Use local Azure IP address.')
+  useLocalAzureIpAddress: bool?
+  
+  @description('Optional. Use policy-based traffic selectors.')
+  usePolicyBasedTrafficSelectors: bool?
+  
+  @description('Optional. VPN connection protocol type.')
+  vpnConnectionProtocolType: ('IKEv1' | 'IKEv2')?
+  
+  @description('Optional. IPSec policies.')
+  ipsecPolicies: array?
```

Comment:
Please import the typs of the `vpn-connection` child-module on top of this type and then reference it like `ipsecPoilicies: ipsecPolicyType[]?`

You can find the syntax for the import in several other modules for reference :) The same comment applies to all other parameters that have types defined in the child modules